hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -83,22 +83,31 @@ public SparkRDDWriteClient(HoodieEngineContext context, HoodieWriteConfig client
 
   @Deprecated
   public SparkRDDWriteClient(HoodieEngineContext context, HoodieWriteConfig writeConfig, boolean rollbackPending) {
-    super(context, writeConfig);
+    this(context, writeConfig, Option.empty());
   }
 
   @Deprecated
   public SparkRDDWriteClient(HoodieEngineContext context, HoodieWriteConfig writeConfig, boolean rollbackPending,
                              Option<EmbeddedTimelineService> timelineService) {
-    super(context, writeConfig, timelineService);
+    this(context, writeConfig, timelineService);
   }
 
   public SparkRDDWriteClient(HoodieEngineContext context, HoodieWriteConfig writeConfig,
                              Option<EmbeddedTimelineService> timelineService) {
     super(context, writeConfig, timelineService);
+    bootstrapMetadataTable(Option.empty());
+  }
+
+  private void bootstrapMetadataTable(Option<String> instantInProgressTimestamp) {
```

Comment:
not sure if we can call it as bootstrap metadata table. this is essentially initializing metadata table. internal impl could do bootstrap if applicable.

Reply:
renamed to initializeMetadataTable